### ISSUE TITLE
[AlertDialog] MaterialAlertDialog has wrong title textSize

### ISSUE THREAD

**Issue by p-dmitriy** [material-components/material-components-android#3542]
**Description:** MaterialAlertDialog has smaller text size than specified in m3 documentation.
![telegram-cloud-photo-size-2-5420222584542317162-y](https://github.com/material-components/material-components-android/assets/82445071/7c90ce1a-a714-42ad-86a5-bbc7cc09c9cd)

**Expected behavior:** Title expected to have `24sp` text size, but actually it is `18.2sp`. LayoutInspector refer to `values-v21.xml` where I don't see 18sp text size. In this file it is also 24sp.
<img width="459" alt="image" src="https://github.com/material-components/material-components-android/assets/82445071/9c1a62de-d525-4447-9982-9634b04b68af">
<img width="697" alt="image" src="https://github.com/material-components/material-components-android/assets/82445071/7ecce673-9577-41e9-b9c3-7cea280210d4">

**Source code:** 
```kotlin
override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
    return MaterialAlertDialogBuilder(requireContext()).apply {
        setTitle(arguments?.getString(KEY_TITLE))
        setMessage(arguments?.getString(KEY_MESSAGE))
        setPositiveButton("Да") { _, _ -> dismiss() }
        setNegativeButton("Нет") { _, _ -> dismiss() }
    }.create()
}
```
In app theme I do not override dialog theme, it default.

**Android API version:** API 33

**Material Library version:** `1.8.0`/`1.9.0`/`1.10.0-alpha06`/`1.11.0-alpha02`

**Device:** Xiaomi 13

---
I also noticed that dialog from 1.11.0-aplha02 has different background color than on 1.10.0-alpha06 and below. Screenshot at the top of this issue is taken with material 1.11.0-aplha02. Background color have 'purplish' tint, but the primary color of theme is a blue (#007AFF). Dialog with material 1.10.0-alpha06 has correct (as I think) 'bluish' color. As I know dialog background color should be surface + elevation overlay, but seems on latest version it is not correct.
![telegram-cloud-photo-size-2-5420222584542317199-y](https://github.com/material-components/material-components-android/assets/82445071/6b3e9f8d-c647-4ecb-92cf-72fe5226c171)

**Comment by bubbleguuum** [comment]
For the dialog background this is expected and due to the tonal surface color changes. Refer to the updated documentation for `Dialog`, the background color is `?attr/colorSurfaceContainerHigh`. I really prefer it this way as I found it the dialog background too much tinted with `colorPrimary` previously.

**Comment by p-dmitriy** [comment]
Ok, documentation really says to use surface container high color. Then I have another question: does tonal elevation needed anymore? Dialog still have level 3 elevation specified. Why do we need to specify elevation if it has no effect for dialog color?

**Comment by imhappi** [comment]
Hi @p-dmitriy ! From my testing on the catalog app, it looks like it is 24sp to me. I also got 24 with the layout inspector

Do you have the device font scale at 100%? A smaller font scale could change the text size to a value with a decimal point, which is very unlikely to be a resource value otherwise

Re: your other question, elevation is still needed for depth/shadows

**Comment by p-dmitriy** [comment]
Yes, I have smaller text size than default (on my phone it's not a percents, it's XS, S, M etc.). The default is S, I use XS. But LayoutInspector shows size in `sp`, which have same value for different sizes. When I switched back to S — I still get 18.2sp in LayoutInspector:
<img width="342" alt="image" src="https://github.com/material-components/material-components-android/assets/82445071/3bc5c3a4-58f6-414e-9515-89a71ba0398d">
And when I set title text style explicitly to HeadlineSmall:
```xml
<item name="materialAlertDialogTitleTextStyle">@style/TextAppearance.Material3.HeadlineSmall</item>
```
I get correct 24sp in LayoutInspector for both XS and S scalings.
Can it be caused by I am using M2 theme for app and for dialog I pass context wrapped with M3 theme with ContextThemeWrapper?
